zookeeper-server/src/test/java/org/apache/zookeeper/server/ZooKeeperServerTest.java

```diff
@@ -135,4 +138,32 @@ public void testInvalidSnapshot() {
         }
     }
 
+    @Test
+    public void testClientZxidAhead() {
+        ZooKeeperServer zooKeeperServer = new ZooKeeperServer();
+        final ZKDatabase zkDatabase = new ZKDatabase(mock(FileTxnSnapLog.class));
+        zooKeeperServer.setZKDatabase(zkDatabase);
+
+        final ByteBuffer output = ByteBuffer.allocate(30);
+        // serialize a connReq
+        output.putInt(1);
+        // lastZxid
+        output.putLong(99L);
+        output.putInt(500);
+        output.putLong(123L);
+        output.putInt(1);
+        output.put((byte) 1);
+        output.put((byte) 1);
+        output.flip();
+
+        try {
+            final NIOServerCnxn nioServerCnxn = mock(NIOServerCnxn.class);
+            zooKeeperServer.processConnectRequest(nioServerCnxn, output);
+        } catch (Exception e) {
+            // expect
+            assertTrue(e instanceof NIOServerCnxn.CloseRequestException);
+            assertEquals(((NIOServerCnxn.CloseRequestException) e).getReason(), ServerCnxn.DisconnectReason.CLIENT_ZXID_AHEAD);
+        }
```

Comment:
I believe there's an `assertThrows()` method that might be cleaner than try-catch, and which returns the exception thrown, so you can check additional state, like the `getReason()`. The try-catch syntax is fine, though.

Reply:
Btw is look good to me even in this form :)

Reply:
> I believe there's an `assertThrows()` method that might be cleaner than try-catch, and which returns the exception thrown, so you can check additional state, like the `getReason()`. The try-catch syntax is fine, though.

Thank you for your advice. 
I have done as you suggested.